installer/secrets.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright The Guard Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package installer
+
+import (
+	"io/ioutil"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func newProxySecret(namespace string, httpsProxy string, httpProxy string, noProxy string) runtime.Object {
+
+	return &core.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "guard-proxy",
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Type: core.SecretTypeOpaque,
+		StringData: map[string]string{
```

Comment:
I think this should also be data. Even if secret takes care to convert that to b64 during creation, if we keep this as string data, on client side it will be in plain text in installer yaml for basic authentication.

Reply:
done